initial ThinningAlgorithm created

ThopClayton opened the pull request:
added a new variable to tree, "dist", which is used by the thinning algorithm.
Also added setter for cut since it appeared to be missing

Created a new class "SortByDist" which is needed to run the thinningAlgorithm

modified class "Tree"'s ToString method: made it so you can now see the "cut" variable and it's value

Added "safeDistance" variable to forest. It is the distance deemed safe between trees. Hard set to 5 right now for testing. Need to implement a way for user to specify (perhaps from file input? GUI?)

Added "ThinningAlgorithm" method to Forest. It marks every tree to be cut or not based on the "safeDistance" variable. I stepped through it a couple of times to make sure it works how it should, and while I do want to test it a lot more, it seems correct from what I've seen.

ThopClayton committed:
initial ThinningAlgorithm created

added a new variable to tree, "dist", which is used by the thinning algorithm.
Also added setter for cut since it appeared to be missing

Created a new class "SortByDist" which is needed to run the thinningAlgorithm

modified class "Tree"'s ToString method: made it so you can now see the "cut" variable and it's value

Added "safeDistance" variable to forest. It is the distance deemed safe between trees. Hard set to 5 right now for testing. Need to implement a way for user to specify (perhaps from file input? GUI?)

Added "ThinningAlgorithm" method to Forest. It marks every tree to be cut or not based on the "safeDistance" variable. I stepped through it a couple of times to make sure it works how it should, and while I do want to test it a lot more, it seems correct from what I've seen.

ThopClayton committed:
added exportToFile method

ThopClayton committed:
Delete .~lock.trimmedTrees.csv#

it was a mistake

Noah-Schoonover committed:
Merge branch 'main' into Thinning-Algorithm

Noah-Schoonover commented on src/main/java/com/treemoval/data/Forest.java:
```diff
@@ -22,6 +21,7 @@
 public class Forest {
 
     List<Tree> trees = new ArrayList<>();
+    private double safeDistance = 5; //need to implement way for user to modify
```
In IntelliJ, if you put `// todo ...blah blah` (for example `// todo need to implement a way for the user to modify this`), the IDE will mark the text yellow, and it will be tagged alongside your scroll bar. Pretty useful stuff, especially since we're all using IntelliJ.

ThopClayton replied:
noted, and added

Noah-Schoonover reviewed (changes requested):
Please change the `cut` member in the `Tree` class to an enumeration. Perhaps we should name this `enum TreeTags` or even just `enum Tags`. Ultimately, it's up to you. However, the enumeration should contain `MARKED` and `UNMARKED` (or `CUT` and `UNCUT` if you prefer). Upon instantiation, every tree should have `Tags tag = UNMARKED;`.

It's better to use constants than plain integers (such as 0 and 1), and enumerations are perfect here. Coding rule of thumb: "[Never use Magic Numbers](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad)"

Also, this isn't detrimental, but try to work on every feature in a different branch (hopefully with its own issue). You added the `exportToFile` method in this branch, which might make it harder to find relevant commits/merges/discussions/etc later on.

Please comment on this review! We should get in the habit of using these issues and pull requests to track and discuss changes.

ThopClayton committed:
added enum for marking trees, removed exportToFile method

Tree and Forest classes were modified to accommodate the shift from overworking integers to utilizing an enumeration.

exportToFile method was removed, Josiah will push his version to a new branch.

ThopClayton commented:
added an enum to avoid overworking integers. Types are UNMARKED, CUT and SAFE. Forest's Thinning Algorithm method and the Tree class were modified to accommodate this new change.

The exportToFile method was removed from this branch for clarification. Josiah should create a new branch for that method very soon.

Noah-Schoonover committed:
Merge branch 'main' into Thinning-Algorithm

Noah-Schoonover committed:
Merge branch 'Thinning-Algorithm' of https://github.com/Noah-Schoonover/cse326-final-project-treemoval into Thinning-Algorithm

Noah-Schoonover reviewed (approved):
Looks good; thanks for moving the export method to a new branch.
